### PR TITLE
Version 3.2.8

### PR DESCRIPTION
**Pull request by yasen-mailjet**
- multi-store owners can now segment their customers by store
- various JS fixers
- added links to User guide and Support at the plugin footer
- fixed marketing triggers cron script
- on creation of a new customer, the email and properties are now synced
properly to Mailjet

### CODE REVIEW

File: classes/Segmentation.php [hunk 1]
```diff
@@ -891,20 +896,20 @@ public function getQuery($post, $live, $limit = false, $speField = '')
 			}
 		}
 
+
+			if($post['sourceSelect'][0] == 4){
+				$multistoreWhere = ' AND c.id_shop = ' . $post['fieldSelect'][0];
```
Comment (Quetzacoalt91) [inline classes/Segmentation.php]: Can you please sanitize the data inserted in the SQL request with an `(int)` cast ?

File: mailjet.php [hunk 2]
```diff
@@ -701,7 +701,9 @@ public function loadConfiguration()
 			&& Db::getInstance()->Execute('INSERT INTO `'._DB_PREFIX_."mj_sourcecondition` VALUES
 				(1, 1, 1, 'LEFT JOIN `%1orders` o ON c.`id_customer` = o.`id_customer`\r\nLEFT JOIN `%1order_detail` od ON o.`id_order` = od.`id_order`\r\nLEFT JOIN `%1currency` cu ON cu.`id_currency` = o.`id_currency`'),
 				(2, 1, 0, 'LEFT JOIN `%1address` ad ON c.`id_customer` = ad.`id_customer` '),
-				(3, 1, 90, NULL)");
+				(3, 1, 90, NULL),
+				(4, 1, 107, 'LEFT JOIN `%1shop` s ON s.`id_shop` = c.`id_shop`')
```
Comment (Quetzacoalt91) [inline mailjet.php]: If you add this line from the version 3.2.8 of the module, you may need to add this line in an upgrade file for modules already installed ?

File: views/templates/admin/configuration.tpl [hunk 3]
```diff
@@ -68,8 +69,7 @@
     <div class="mj_landing">
         <div align="center">
             <br />
-            <a href="https://www.mailjet.com/support"
-               target="_blank">{l s='If you have a question or if you have a problem, click here to contact support.' mod='mailjet'}</a>
+            {{l s='If you have a question or if you have a problem, click here to contact support.' mod='mailjet'}|html_entity_decode}
```
Comment (Quetzacoalt91) [inline views/templates/admin/configuration.tpl]: I guess that your `html_entity_decode` should be removed.
```php
{l s='If you have a question or if you have a problem, click here to contact support.' mod='mailjet'}
```
would be enough

File: mailjet.php (hunk 2 above)
Comment (yasen-mailjet) [inline mailjet.php]: @Quetzacoalt91 is there any documentation that shows how to do it? Thank you.

Comment (Quetzacoalt91) [inline mailjet.php]: Of course, here it is: http://doc.prestashop.com/display/PS15/Auto-updating+modules

File: views/templates/admin/configuration.tpl (hunk 3 above)
Comment (yasen-mailjet) [inline views/templates/admin/configuration.tpl]: @Quetzacoalt91 it turns out that without the decoding the text shows as:
```javascript
Should you have any questions or encounter any difficulties, please consult our <a target="_blank" href="https://www.mailjet.com/guides/prestashop-user-guide"><strong>User Guide</strong></a> or contact our technical <a target="_blank" href="https://www.mailjet.com/support/ticket"><strong>Support Team</strong></a>
```
will put back the decoding if you don't mind...

Comment (Quetzacoalt91) [inline views/templates/admin/configuration.tpl]: Oh, so you wrote HTML code in a translated text. That's tricky.

You can guess there is a reason if the outputed text is escaped. :wink:

Comment (yasen-mailjet) [inline views/templates/admin/configuration.tpl]: Updated the pull request. Yea there's got to be a reason for the encoding :)

Comment (Quetzacoalt91) [inline views/templates/admin/configuration.tpl]: Hi @yasen-mailjet,

Sorry for this late answer. If you really want to add HTML code in your translations, here is an example you can reuse: https://github.com/PrestaShop/PrestaShop/blob/6c9365135d455938a266461728b346ed43b54b2f/admin-dev/themes/default/template/controllers/backup/helpers/list/list_header.tpl#L71
But in this case, you cannot have a URL for each language.

**Comment by Quetzacoalt91** [comment]
Hi @yasen-mailjet,

Your translation files are incorrect, that's why they are not working. Because you changes the original text, the key in your array is not the same anymore.

I tried to regenerate it and it is working :)
capture du 2015-09-23 09 28 25

**Comment by yasen-mailjet** [comment]
Hi @Quetzacoalt91,

Thank you for pointing out my silly mistake. I've fixed it.

**Comment by Quetzacoalt91** [comment]
Hi @yasen-mailjet,

Thanks, your PR hast just been merged and the changes will be available from tomorrow.
For the next release, it would be great to take care of the validator comments. :wink: 

Best regards

**Pull request merged**